mvc/src/main/java/com/interface21/webmvc/servlet/mvc/tobe/ControllerScanner.java

```diff
@@ -0,0 +1,38 @@
+package com.interface21.webmvc.servlet.mvc.tobe;
+
+import com.interface21.context.stereotype.Controller;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import org.reflections.Reflections;
+
+public class ControllerScanner {
+
+    private final Reflections reflections;
+    private final InstanceInitializer instanceInitializer;
+
+    public ControllerScanner(String basePackage) {
+        this.reflections = new Reflections(basePackage);
+        this.instanceInitializer = new InstanceInitializer();
+    }
+
+    public Map<Class<?>, Object> getControllers(ScanType scanType) throws ReflectiveOperationException {
+        Set<Class<?>> classes = new HashSet<>();
+        switch (scanType) {
+            case ScanType.ANNOTATION -> classes.addAll(reflections.getTypesAnnotatedWith(Controller.class));
+            case ScanType.SUB_TYPE_HIERARCHY ->
+                    classes.addAll(reflections.getSubTypesOf(com.interface21.webmvc.servlet.mvc.asis.Controller.class));
+            default -> throw new IllegalArgumentException("존재하지 않는 Scan Type입니다.");
+        }
+        return instantiateControllers(classes);
+    }
```

Comment:
[제안]
지금의 switch 문은 if문과 다를게 없어보여요 🤔
좀더 가독성이 좋게 하기 위해 switch 문을 고려하셨다면 아래처럼 수정해보는 것은 어떨까요 ?

```suggestion
classes.addAll(switch (scanType) {
    case ANNOTATION -> reflections.getTypesAnnotatedWith(Controller.class);
    case SUB_TYPE_HIERARCHY -> reflections.getSubTypesOf(com.interface21.webmvc.servlet.mvc.asis.Controller.class);
    default -> throw new IllegalArgumentException("존재하지 않는 Scan Type입니다.");
});
```

Reply:
오호 더 깔끔한 코드가 되겠네요!
반영했습니다!